x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java

```diff
@@ -600,6 +623,25 @@ public void testInvalidateApiKeysOwnedByCurrentAuthenticatedUser() throws Interr
         verifyInvalidateResponse(noOfApiKeysForUserWithManageApiKeyRole, userWithManageApiKeyRoleApiKeys, invalidateResponse);
     }
 
+    public void testInvalidateApiKeysOwnedByRunAsUser() throws InterruptedException, ExecutionException {
+        int noOfSuperuserApiKeys = randomIntBetween(3, 5);
+        int noOfApiKeysForUserWithManageApiKeyRole = randomIntBetween(3, 5);
+        createApiKeys(noOfSuperuserApiKeys, null);
+        final String userWithManageOwnApiKeyRole = "user_with_manage_own_api_key_role";
+        final String userWithRunAsRole = "user_with_run_as_role";
+        List<CreateApiKeyResponse> userWithManageApiKeyRoleApiKeys = createApiKeys(userWithManageOwnApiKeyRole,
+            userWithRunAsRole, noOfApiKeysForUserWithManageApiKeyRole, null, "monitor");
+        final Client client = client().filterWithHeader(Map.of("Authorization", UsernamePasswordToken
+            .basicAuthHeaderValue(userWithRunAsRole, SecuritySettingsSourceField.TEST_PASSWORD_SECURE_STRING),
+            "es-security-runas-user", userWithManageOwnApiKeyRole));
+
+        PlainActionFuture<InvalidateApiKeyResponse> listener = new PlainActionFuture<>();
+        client.execute(InvalidateApiKeyAction.INSTANCE, InvalidateApiKeyRequest.forOwnedApiKeys(), listener);
```

Comment:
same as above, we should also test without the `owner` flag toggled but with the concrete `username` and `realm` set.

_I would also add a negative test, but this is usually better to do in unit tests, but I sometimes sneak in a negative test in integ tests as well._

Reply:
Added negative tests for calling [get](https://github.com/elastic/elasticsearch/pull/52178/files#diff-f1e82384e2498269e87f0e84280c8dbcR578) and [invalidate](https://github.com/elastic/elasticsearch/pull/52178/files#diff-f1e82384e2498269e87f0e84280c8dbcR684) API key with mismatching username and/or realm name.